### PR TITLE
feat(core): implement temperature sampling

### LINKED ISSUE #241: core: Implement `TemperatureSample` postprocessing step
## Context

`PostprocessingStep::TemperatureSample` is stubbed at `crates/xybrid-core/src/execution/postprocessing/mod.rs:60`. This is needed for non-llama.cpp logit-output models (small classification heads, custom LMs run via ONNX) where we want to sample rather than take argmax. Medium difficulty — wants someone comfortable with sampling primitives but doesn't need Xybrid internals.

## Scope

- Implement temperature scaling (`logits / temperature`).
- Implement top-k filtering.
- Implement top-p (nucleus) filtering.
- Final multinomial sample via a seedable RNG so tests are deterministic.
- Accept all three parameters via the step's config.

## Out of scope

- Beam search, repetition penalties, frequency penalties, or other advanced sampling tricks.
- Modifying llama.cpp's existing sampler path.

## Acceptance criteria

- [ ] Unit tests verify: temperature=0 ≈ argmax, top-k=1 ≈ argmax, top-p=1.0 + high temp ≈ uniform over allowed tokens.
- [ ] Determinism with a fixed seed.
- [ ] Clear error on invalid params (negative temp, top_k=0, etc.).

## References

- `crates/xybrid-core/src/execution/postprocessing/mod.rs:60` (TODO marker)

### PR DESCRIPTION
**Pull request by original4422**
## Summary

- wire `TemperatureSample` into postprocessing and validate its existing `temperature`, `top_k`, and `top_p` config
- apply stable temperature scaling, composed top-k/top-p filtering, and a weighted draw using production entropy
- keep fixed-seed RNG injection internal and cover deterministic, statistical, edge, input, dispatcher, and serde behavior

Fixes #241

## Testing

- `cargo fmt --all -- --check`
- `cargo test -p xybrid-core --lib --features ort-download` (1,031 passed, 1 ignored)
- `cargo test --workspace --features ort-download`
- `cargo clippy --workspace --all-targets -- -D warnings`

## Bazel note

`bazelisk test //crates/xybrid-core:xybrid-core_test --test_filter=temperature_sample` resolved dependencies and completed analysis, but no Rust test executed because the unconditional native llama.cpp build failed first. Hermetic Clang 22 rejects `CF_ENUM` declarations from Apple CoreFoundation headers while compiling vendored cpp-httplib (`-Welaborated-enum-base`). Cargo workspace validation, including the native llama/whisper paths, passes.

### CODE REVIEW

File: crates/xybrid-core/src/execution/postprocessing/tensor_ops.rs [hunk 1]
```diff
@@ -348,6 +405,118 @@ pub fn argmax_token(logits: &ArrayD<f32>) -> ExecutorResult<usize> {
...
+
+    let data = tensor.as_slice().ok_or_else(|| {
+        AdapterError::InvalidInput(
+            "TemperatureSample requires a contiguous logits tensor".to_string(),
+        )
+    })?;
+    let vocab_size = *shape.last().ok_or_else(|| {
+        AdapterError::InvalidInput("TemperatureSample received an empty tensor shape".to_string())
+    })?;
+    Ok(&data[..vocab_size])
```
[the remark this answers is not in the document]
Comment (original4422) [reply crates/xybrid-core/src/execution/postprocessing/tensor_ops.rs]: Addressed in e8e5d905. For 3D [batch, seq_len, vocab_size] tensors, exact argmax and positive-temperature sampling now both use the final sequence position of the first batch. Regression coverage exercises both paths. Validation passed: 9 focused TemperatureSample tests, all 18 tensor-op tests, xybrid-core 1,036 passed with 1 ignored, workspace Clippy, and the full ort-download workspace test suite.

File: crates/xybrid-core/src/execution/postprocessing/tensor_ops.rs [hunk 2]
```diff
@@ -105,6 +107,61 @@ pub fn topk_step(data: RawOutputs, k: usize, dim: Option<usize>) -> ExecutorResu
...
+        RawOutputs::TensorMap(map) => map,
+        _ => {
+            return Err(AdapterError::InvalidInput(
+                "TemperatureSample requires tensor map".to_string(),
+            ))
+        }
+    };
+    let tensor = tensor_map.values().next().ok_or_else(|| {
+        AdapterError::InvalidInput("No outputs for TemperatureSample".to_string())
+    })?;
```
[the remark this answers is not in the document]
Comment (original4422) [reply crates/xybrid-core/src/execution/postprocessing/tensor_ops.rs]: Addressed in e8e5d905. TemperatureSample now selects the named logits output when present, falls back only when exactly one output exists, and returns InvalidInput for ambiguous multiple unnamed outputs. Regression coverage includes named logits plus a KV-style auxiliary output, the ambiguous multi-output error, and the single-output fallback. The focused, crate, workspace Clippy, and full workspace test validations all pass.

**Review by theGlenn — APPROVED** [review]
thanks for your contribution @original4422. 
Looks good to me
